packages/babel-parser/src/parser/expression.js

```diff
@@ -704,6 +699,69 @@ export default class ExpressionParser extends LValParser {
     }
   }
 
+  checkSliceArgument(pos: number, expr: N.Expression) {
+    if (
+      !expr ||
+      expr.type === "NumericLiteral" ||
+      expr.type === "Identifier" ||
+      (expr.type === "UnaryExpression" &&
+        expr.operator === "-" &&
+        (expr.argument.type === "NumericLiteral" ||
+          expr.argument.type === "Identifier"))
+    ) {
```

Comment:
We also need to check that `expr.extra && expr.extra.parenthesized` is false.

Reply:
Correct, dogfooding another stage 3 proposal. 😝